metrics/storage/blogbench.sh

```diff
@@ -0,0 +1,124 @@
+# they are total scores for all iterations (this is the blogbench default output)
+
+set -e
+
+# General env
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+TEST_NAME="blogbench"
+IMAGE="local-blogbench"
+DOCKERFILE="${SCRIPT_PATH}/blogbench_dockerfile/Dockerfile"
+
+# Number of iterations for blogbench to run - note, results are not
+# scaled to iterations - more iterations results in bigger results
+ITERATIONS="${ITERATIONS:-30}"
+
+# Directory to run the test on
+# This is run inside of the container
+TESTDIR="${TESTDIR:-/tmp}"
+CMD="blogbench -i ${ITERATIONS} -d ${TESTDIR}"
+
+function main() {
+	# Check tools/commands dependencies
+	cmds=("awk" "docker")
+
+	init_env
+	check_cmds "${cmds[@]}"
+	check_dockerfiles_images "$IMAGE" "$DOCKERFILE"
+
+	metrics_json_init
+
+	local output=$(docker run --rm --runtime=$RUNTIME $IMAGE $CMD)
+
+	# Save configuration
+	metrics_json_start_array
+
+	local frequency=$(echo "$output" | grep "Frequency" | cut -d "=" -f2 | cut -d ' ' -f2)
+	local iterations=$(echo "$output" | grep -w "iterations" | cut -d ' ' -f3)
+	local spawing_writers=$(echo "$output" | grep -w "writers" | cut -d ' ' -f2)
+	local spawing_rewriters=$(echo "$output" | grep -w "rewriters" | cut -d ' ' -f2)
+	local spawing_commenters=$(echo "$output" | grep -w "commenters" | cut -d ' ' -f2)
+	local spawing_readers=$(echo "$output" | grep -w "readers" | cut -d ' ' -f2)
+
+	local json="$(cat << EOF
+	{
+		"Frequency" : $frequency,
+		"Iterations" : $iterations,
+		"Number of spawing writers" : $spawing_writers,
+		"Number of spawing rewriters" : $spawing_rewriters,
+		"Number of spawing commenters" : $spawing_commenters,
+		"Number of spawing readers" : $spawing_readers
+	}
+EOF
+)"
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Config"
+
+	# Save results
+	metrics_json_start_array
+
+	local writes=$(tail -2 <<< "$output" | head -1 | awk '{print $5}')
+	local reads=$(tail -1 <<< "$output" | awk '{print $6}')
+
+	# Obtaining other Blogbench results
+	local nb_blogs=$(echo "$output" | tail -n +12 | head -n -3 | awk ' BEGIN {ORS="\t"} {print $1} ' | tr '\t' ',' | sed '$ s/.$//')
+	local r_articles=$(echo "$output" | tail -n +12 | head -n -3 | awk ' BEGIN {ORS="\t"} {print $2} ' | tr '\t' ',' | sed '$ s/.$//')
+	local w_articles=$(echo "$output" | tail -n +12 | head -n -3 | awk ' BEGIN {ORS="\t"} {print $3} ' | tr '\t' ',' | sed '$ s/.$//')
+	local r_pictures=$(echo "$output" | tail -n +12 | head -n -3 | awk ' BEGIN {ORS="\t"} {print $4} ' | tr '\t' ',' | sed '$ s/.$//')
+	local w_pictures=$(echo "$output" | tail -n +12 | head -n -3 | awk ' BEGIN {ORS="\t"} {print $5} ' | tr '\t' ',' | sed '$ s/.$//')
+	local r_comments=$(echo "$output" | tail -n +12 | head -n -3 | awk ' BEGIN {ORS="\t"} {print $6} ' | tr '\t' ',' | sed '$ s/.$//')
+	local w_comments=$(echo "$output" | tail -n +12 | head -n -3 | awk ' BEGIN {ORS="\t"} {print $7} ' | tr '\t' ',' | sed '$ s/.$//')
```

Comment:
Please can you either use variables for all these magic numbers or (better) create a new variable so you can say something like:

```
local -r data=$(echo "$output" | tail -n +12 | head -n -3)

local nb_blogs=$(echo "$data" | awk ' BEGIN {ORS="\t"} {print $1} ' | tr '\t' ',' | sed '$ s/.$//') )
local r_articles=(echo "$data" | awk ' BEGIN {ORS="\t"} {print $2} ' | tr '\t' ',' | sed '$ s/.$//') )
  :
```

In fact, I wonder if you can simplify this even further to avoid each variable also calling `tr` and `sed`?